src/github/amexboy/project_gen.clj

```diff
@@ -0,0 +1,129 @@
+  (some (partial = "pom.xml") (.list dir))
+  )
+(defn gradle-root? [dir]
+  "Checks if the dir contains a maven.pom file"
+  (some (partial = "build.gradle") (.list dir))
+  )
+(defn liengen-root? [dir]
+  "Checks if the dir contains a maven.pom file"
+  (some (partial = "project.clj") (.list dir))
+  )
+
+(defn gen-maven-iml [path name]
+  "Generates an IDEA module for the path with the given name"
+  (let [template "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+           <module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule=\"true\" type=\"JAVA_MODULE\" version=\"4\">
+           <component name=\"ExternalSystem\" externalSystem=\"Maven\" />
+        </module>"]
+    (spit (str path "/" name ".iml") template))
+  )
+
+(defn gen-gradle-iml [path name]
+  "Generates an IDEA module for the path with the given name"
+  (let [template "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+           <module org.jetbrains.idea.gradle.project.GradlenProjectsManager.isGradleModule=\"true\" type=\"JAVA_MODULE\" version=\"4\">
+           <component name=\"ExternalSystem\" externalSystem=\"Maven\" />
+        </module>"]
+    (spit (str path "/" name ".iml") template))
+  )
+
+(defn gen-liengen-iml [path name]
+  "Generates an IDEA module for the path with the given name"
+  (let [
+        template "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+                     <module cursive.leiningen.project.LeiningenProjectsManager.displayName=\"" name "\"
+                             cursive.leiningen.project.LeiningenProjectsManager.isLeinModule=\"true\"
+                             type=\"JAVA_MODULE\" version=\"4\">
+                     <component name=\"ExternalSystem\" externalSystem=\"Maven\" />
+                  </module>"]
+    (spit (str path "/" name ".iml") template))
+  )
+
+(defn gradle-entry [module] 
+  (str "
+  <component name=\"GradleSettings\">
+    <option name=\"linkedExternalProjectsSettings\">
+      <GradleProjectSettings>
+        <option name=\"distributionType\" value=\"DEFAULT_WRAPPED\" />
+        <option name=\"externalProjectPath\" value=\"$PROJECT_DIR$/" module "/\" />
+        <option name=\"useQualifiedModuleNames\" value=\"true\" />
+      </GradleProjectSettings>
+    </option>
+ </component>
+  ")
+  )
+
+(defn module-entries [root module]
+  "Returns component entries for git, pom (gradle soon) and git component lists"
+  (let [entries {
+                 :iml (str "<module filepath=\"file://$PROJECT_DIR$/" module "/" module ".iml\"")
+                 :git (str "<mapping directory=\"$PROJECT_DIR$/" module "\" vcs=\"Git\" />")
+                 }
+        module-file (io/file root module)]
+    (cond
+      (maven-root? module-file) (assoc entries :pom (str "<option value=\"$PROJECT_DIR$/" module "/pom.xml\" />"))
+      (gradle-root? module-file) (assoc entries :gradle (gradle-entry module))
+      ))
+  )
+
+(defn reducer [r v]
+  (reduce #(assoc %1 %2 (cons (%2 r) (%2 v)))
+          r (keys v)))
```

Comment:
This is pretty dense to use the simple anonymous function syntax on, I would use `fn` and give the variable names and use longer names for `r` and `v`.

This can also be re-written more simply I think.  It looks like this is just taking two maps that are expected to have the same set of keys and merging their values by concatenating them, so you could write this as:

`(merge cons r v)`

Also, I think you want to use `conj` instead.  In general, I never use `cons` because `conj` does exactly the same thing when given a list, but is more general and idiomatic.  Also, since the argument order is more consistent that is also nice.